content/socials.js

```diff
@@ -67,23 +74,44 @@ export default class Socials {
     const matcher = /(?:https?:\/\/)?(?:www.)?facebook.(?:[a-z]+)\/([^/?\s]+)/i;
 
     return {
-      matcher
+      match: (input) => {
+        if (!matcher.test(input)) {
+          return null;
+        }
+
+        const matches = matcher.exec(input);
+        return matches[1];
+      }
     };
   }
 
   static get twitter () {
     const matcher = /(?:https?:\/\/)?(?:www.)?twitter.(?:[a-z]+)\/([^/?\s]+)/i;
 
     return {
-      matcher
+      match: (input) => {
+        if (!matcher.test(input)) {
+          return null;
+        }
+
+        const matches = matcher.exec(input);
+        return matches[1];
+      }
     };
   }
 
   static get reddit () {
     const matcher = /(?:https?:\/\/)?(?:www.)?reddit.(?:[a-z]+)\/u\/([^/?\s]+)/i;
 
     return {
-      matcher
+      match: (input) => {
+        if (!matcher.test(input)) {
+          return null;
+        }
+
+        const matches = matcher.exec(input);
```

Comment:
I think I thought it was faster to check with `test` (so better if little matches are expected). There's also the use of the `g` flag to could alter the `exec` results.